problem-gson/src/test/java/org/zalando/problem/ProblemAdapterFactoryTest.java

```diff
@@ -0,0 +1,40 @@
+package org.zalando.problem;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.ThrowingSupplier;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class ProblemAdapterFactoryTest {
+
+    @Test
+    void defaultConstructorShouldBuildIndexCorrectly() {
+        assertDoesNotThrow((ThrowingSupplier<ProblemAdapterFactory>) ProblemAdapterFactory::new);
+    }
+
+    @Test
+    void shouldThrowForDuplicateStatusCode() {
+        assertThrows(IllegalArgumentException.class, () -> new ProblemAdapterFactory(Status.class, CustomStatus.class));
+    }
+
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    @Test
+    void shouldThrowForDuplicateSubtype() {
```

Comment:
That one should work, shouldn't it?

Reply:
I mean it should be possible to register two different types to the same implementation class.

Reply:
Yes, should be possible to tie multiple types to the same implementation class. I'll do the required changes 👍